pkg/mtls/certwatch/tls_config_holder.go

```diff
@@ -24,9 +25,10 @@ type TLSConfigHolder struct {
 }
 
 // NewTLSConfigHolder instantiates a new instance of TLSConfigHolder
-func NewTLSConfigHolder(rootCfg *tls.Config) *TLSConfigHolder {
+func NewTLSConfigHolder(rootCfg *tls.Config, fallbackClientAuthNoCAs tls.ClientAuthType) *TLSConfigHolder {
```

Comment:
If kept as-is, we should rename this to `defaultClientAuth` or `fallbackClientAuth` with a clarification in the comment how the interaction is currently working.